### ISSUE TITLE
key-spacing with align doesn't allow mixed styles

### ISSUE THREAD

**Issue by mourner** [eslint/eslint#3938]
This issue continues the discussion in #1728. Here is a use case I'd love to see supported:

```js
// eslint key-spacing: [2, {"align": "value", "mixed": true}]

// both aligned and non-aligned groups are accepted:
var foo = {
    // not aligned
    first: 1,
    second: 2,

    // aligned
    third:  3,
    fourth: 4
}

// but you can't mix styles in a block like this:
var foo = {
	first: 1,
	second: 2,
	third:  3,
	fourth: 4
};
```

The lack of support for this has been preventing me from using `key-spacing` rule in most of [my projects](https://github.com/mourner/projects), and I think the case is much more common than what `align` option currently enforces. 

The most common style I've seen in JS projects is not aligning properties in general but with exceptions where it makes the code nicer (e.g. values have similar lines that line up well). Here's a [real world example](https://github.com/Leaflet/Leaflet/blob/master/src/dom/DomEvent.Pointer.js).

The proposed `mixed: true` option could be equivalently done as `requireAlignment: false`.

cc @btmills @nzakas @moll @doberkofler @mayatron

#2668 is related but not the same issue.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by Scotchester** [comment]
:+1: Yes, please!

**Comment by gyandeeps** [comment]
@eslint/eslint-team thoughts here?

**Comment by nzakas** [comment]
Up to @btmills

**Comment by btmills** [comment]
Let's go with the `requireAlignment` option. Fair warning to whoever wants to work on this: the rule is already very complex in order to handle all of the different options. Before you submit a PR, I recommend spending some time in advance on your design, and feel free to discuss it here to get feedback.

**Comment by doberkofler** [comment]
Sounds good to me

**Comment by alberto** [comment]
Closed by lack of activity
